src/crepe/util/OptionalRef.hpp

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <stdexcept>
+
+#include "OptionalRef.h"
+
+namespace crepe {
+
+template <typename T>
+OptionalRef<T>::OptionalRef(T & ref) {
+	this->set(ref);
```

Comment:
Kan `set()` niet gewoon weg? Het lijkt mij overbodig om hier een functieaanroep te doen naar een functie die letterlijk alleen `this->ref = &ref` doet.